Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs

```diff
@@ -113,49 +114,76 @@ public static class WebAppServiceCollectionExtensions
         /// <returns></returns>
         public static IServiceCollection AddMsal(this IServiceCollection services, IConfiguration configuration, IEnumerable<string> initialScopes, string configSectionName = "AzureAd")
         {
+            // Ensure that configuration options for MSAL.NET, HttpContext accessor and the Token acquisition service
+            // (encapsulating MSAL.NET) are available through dependency injection
             services.Configure<ConfidentialClientApplicationOptions>(options => configuration.Bind(configSectionName, options));
-
             services.AddHttpContextAccessor();
             services.AddTokenAcquisition();
 
-            services.Configure<OpenIdConnectOptions>(AzureADDefaults.OpenIdScheme, options =>
+            services.Configure(AzureADDefaults.OpenIdScheme, (System.Action<OpenIdConnectOptions>)(options =>
+            {
+                SetBasicMsalConfigurations(initialScopes, options);
+            }));
+
+            return services;
+        }
+
+        /// <summary>
+        /// Add MSAL support to the B2C Web App or B2C Web API
+        /// </summary>
+        /// <param name="services">Service collection to which to add authentication</param>
+        /// <param name="initialScopes">Initial scopes to request at sign-in</param>
+        /// <returns></returns>
+        public static IServiceCollection AddMsalB2C(this IServiceCollection services, IConfiguration configuration, IEnumerable<string> initialScopes, string configSectionName = "AzureAdB2C")
```

Comment:
can we discuss changing this and also .AddMsal w/@jmpriuer. i know we have an issue about naming, so a good time to bring this up. We shouldn't have any B2C branding, so how can we differentiate B2C from AAD?